persistence/proto/src/main/scala/net/liftweb/proto/ProtoUser.scala

```diff
@@ -864,8 +864,10 @@ trait ProtoUser {
       }
     }
 
+    val emailElemId = nextFuncName
+    S.appendJs(Focus(emailElemId))
     val bind =
-      ".email" #> FocusOnLoad(<input type="text" name="username"/>) &
+      ".email" #> <input type="text" name="username" id={emailElemId}/> &
```

Comment:
I don't actually think using attribute extraction is an option here, as there's no event attribute we can attach this behavior to. I mean, we can always update the attribute extraction code to see `autofocus` and turn it into JS, but that's a counter-goal IMO.

In short, I think the current solution you have is correct :)